glue_vispy_viewers/isosurface/layer_artist.py

```diff
@@ -13,15 +15,80 @@
 from glue.core.exceptions import IncompatibleAttribute
 
 
+from ..extern.vispy.color import BaseColormap, get_colormaps
+from ..extern.vispy.visuals.volume import frag_dict, FRAG_SHADER
+
+from itertools import cycle
+from .multi_iso_visual import MultiIsoVisual
+
+
+# COLOR_BREWER = None
+BREWER_COLOR_FILE = os.path.join(os.path.dirname(__file__), 'ColorBrewer_all_schemes_RGBonly3.XLS')
+
+
+# TODO: add parameters of cmap and level from IsosurfaceLayerArtist
+def set_brewer_color():
+    import pandas as pd
+
+    xl = pd.ExcelFile(BREWER_COLOR_FILE)
+    df = xl.parse('Sheet1')[:1690]
+
+    # TODO: 4.0 here will be replaced by level, 'Blues' will be replaced by chosen color in dropdown list
+    # select specific data series upon UI setting
+    mask = df['ColorName'].isin(['Blues']) & df['NumOfColors'].isin([4.0]) & df['Type'].isin(['seq'])
+    sel = df[mask]
+    print('sel', sel)
+    index = sel.index[0] # int
+    df_rgb = df.loc[range(index, index+int(4.0), 1)]
+
+    # get color array
+    color = np.ones((int(4.0), 4))
+    color[:, 0] = np.array(df_rgb['R'])
+    color[:, 1] = np.array(df_rgb['G'])
+    color[:, 2] = np.array(df_rgb['B'])
+    color[:, 3] = 255.0
+    color /= 255.
+    return color.tolist()
+
+
+class Brewer(BaseColormap):
+    colors = set_brewer_color()
+
+    # Use $color_0 to refer to the first color in `colors`, and so on. These are vec4 vectors.
+    glsl_map = """
+    vec4 translucent_grays(int l) {
+        if (l == 1)
+            {return $color_0;}
+        if (l == 2)
+            {return $color_1;}
+        if (l == 3)
+            {return $color_2;}
+        if (l == 4)
+            {return $color_3;}
+    }
+    """
+
```

Comment:
@astrofrog How can I enable a function( `set_brewer_color` as above) to accept parameters from one class( `IsosurfaceLayerArtist` as below) to track user settings on display, while assign the output of the same function to a parameter in another class (`Brewer` as above) as color lists? 

I tried `global` variable but seems the initiation of two classes are ahead of inner functions ;( Thanks a lot!!!